astore/client/auth/auth.go

```diff
@@ -103,5 +113,42 @@ func (c *Client) Login(username, domain string, o LoginOptions) (string, error)
 		return "", fmt.Errorf("could not decrypt returned token")
 	}
 
+	publicKey, _, _, _, err := ssh.ParseAuthorizedKey(tres.Capublickey)
+	if err != nil {
+
+	}
+	sshDir, err := kcerts.FindSSHDir()
+	if err != nil {
+		return "", err
+	}
+	err = kcerts.AddSSHCAToClient(publicKey, tres.Cahosts, sshDir)
+	if err != nil {
+		return "", err
+	}
+	agent, err := kcerts.FindSSHAgent(o.Store, o.L)
+	if err != nil {
+		return "", err
+	}
+	defer agent.Close()
+	file, err := ioutil.TempFile("/tmp", "en")
+	if err != nil {
+		return "", err
+	}
+	err = marshal.MarshalFile(file.Name(), tres.Key)
```

Comment:
No! Don't write the private key in /tmp! Also, no reason to fork off to `ssh-add`, just use the golang ssh-agent client API directly (add a Connect() method to the SSHAgent object you created in previous PR?). Code could look like this ([a full example here](https://pkg.go.dev/golang.org/x/crypto/ssh/agent#NewClient)):
```
	conn, err := net.Dial("unix", agent.Socket)
        ... err ...
        // from import "golang.org/x/crypto/ssh/agent"
	agentClient := agent.NewClient(conn)
        agentClient.Add(...)
```

Reply:
also... we need something a bit smarter to add the key: we need to garbage collect old keys? Make sure to have a Lifetime specified).

If we don't garbage collect keys, this whole thing will stop working in like 5 days (reason: last time I checked how this worked, the ssh client presented one key at a time to the server, but the server refused to challenge more than X keys - X == 5? to avoid DoS and similar).